sharding/vmc.go

```diff
@@ -30,14 +43,39 @@ func (c *Client) verifyVMC() error {
 	if len(b) == 0 {
 		log.Info(fmt.Sprintf("No validator management contract found at %s.", validatorManagerAddress.String()))
 
-		addr, err := c.deployVMC()
-		if err != nil {
-			return fmt.Errorf("unable to deploy validator management contract: %v", err)
+		accounts := c.keystore.Accounts()
+		if len(accounts) == 0 {
+			return fmt.Errorf("no accounts found")
 		}
 
+		if err := c.unlockAccount(accounts[0]); err != nil {
+			return fmt.Errorf("unable to unlock account 0: %v", err)
+		}
+		ops := bind.TransactOpts{
+			From: accounts[0].Address,
+			Signer: func(signer types.Signer, addr common.Address, tx *types.Transaction) (*types.Transaction, error) {
+				networkID, err := c.client.NetworkID(context.Background())
+				if err != nil {
+					return nil, fmt.Errorf("unable to fetch networkID: %v", err)
+				}
+				return c.keystore.SignTx(accounts[0], tx, networkID)
```

Comment:
EIP155 Signer normally use CHAIN_ID instead of NETWORK_ID.
Maybe a comment to mention the clever hack

Reply:
Done. For whatever reason, the ethclient API references the chain ID as "network ID". They even have a comment to mention "also know as the chain ID" https://github.com/ethereum/go-ethereum/blob/722bac84fa503199b9c485c1a3e2bfba03bc487d/ethclient/ethclient.go#L305